src/Builder.jl

```diff
@@ -104,36 +97,23 @@ function Selectors.runner(::Type{SetupBuildDirectory}, doc::Documents.Document)
             end
             for file in files
                 src = normpath(joinpath(root, file))
-                dst = normpath(joinpath(build, relpath(root, source), file))
+                path = normpath(joinpath(relpath(root, source), file)) # path relative to source
+                dst = normpath(joinpath(build, path))
                 # Non-markdown files are simply copied over to `build`.
                 # Markdown files get added to the document tree as `Page` objects.
-                endswith(src, ".md") ?
-                    Documents.addpage!(doc, src, dst) :
+                if endswith(file, ".md")
+                    name = first(splitext(path))
+                    Documents.addpage!(doc, name, src, dst)
+                else
                     cp(src, dst; remove_destination = true)
+                end
```

Comment:
The more minimal change for this would just be

```diff
diff --git a/src/Documents.jl b/src/Documents.jl
index b26be4d..4ba9395 100644
--- a/src/Documents.jl
+++ b/src/Documents.jl
@@ -132,7 +132,8 @@ end
 
 function addpage!(doc::Document, src::AbstractString, dst::AbstractString)
     page = Page(src, dst)
-    doc.internal.pages[src] = page
+    name, _ = splitext(relpath(src, doc.user.source))
+    doc.internal.pages[name] = page
 end
```

which I feel would be much more preferable.